### PR TITLE
JAMES-3491 Push StateChanges for VacationResponses and EmailDeliveries

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
`MemoryWebAdminServerIntegrationTest>WebAdminServerIntegrationTest.getSwaggerShouldReturnJsonDataForSwagger:309 1 expectation failed.`

Unrelated

test this please

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/WebSocketContract.scala [hunk 1]
```diff
@@ -577,13 +577,137 @@ trait WebSocketContract {
...
 
     assertThat(response.toOption.get.asJava)
       .hasSize(3) // email notification + mailbox notification + API response
       .contains(mailboxStateChange, emailStateChange)
   }
 
+  @Test
+  @Timeout(180)
+  // For client compatibility purposes
+  def emailDeliveryShouldNotIncludeFlagUpdatesAndDeletes(server: GuiceJamesServer): Unit = {
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/WebSocketContract.scala:645]: No test to check if we get a EmailDelivery notification after creating one?

Comment (chibenwa) [reply server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/WebSocketContract.scala:645]: `dataTypesShouldDefaultToAll` already ensure that.

I'm unsure of the added value of a separated test.

**Review by Arsnael — APPROVED** [review]

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**